forte/data/ontology/ontology_code_generator.py

```diff
@@ -811,22 +812,25 @@ def parse_schema(
                 module_writer.add_entry(en, entry_item)
 
             # Adding entry attributes to the allowed types for validation.
-            for property_name in properties:
+            for property in properties:
+                property_name = property[0]
                 # Check if the name is allowed.
                 if not property_name.isidentifier():
                     raise InvalidIdentifierException(
                         f"The property name: {property_name} is not a valid "
                         f"python identifier."
                     )
 
-                if property_name in self.allowed_types_tree[en.class_name]:
+                if property_name in set(
+                    val[0] for val in self.allowed_types_tree[en.class_name]
+                ):
                     warnings.warn(
                         f"Attribute type for the entry {en.class_name} "
                         f"and the attribute {property_name} already present in "
                         f"the ontology, will be overridden",
                         DuplicatedAttributesWarning,
                     )
-                self.allowed_types_tree[en.class_name].add(property_name)
+                self.allowed_types_tree[en.class_name].add(tuple(property))
```

Comment:
Do we need this `tuple` conversion? The type of `property` should already be a `tuple`